TwoLayers_Geometry.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Oct 11 15:56:57 2024
+
+@author: Anupama Rajendra
+"""
+
+import openmc
+
+# Create geometry
+#Spherical shell:
+def make_spherical_shell(inner_radius_W, outer_radius_W, inner_radius_C, M_1, M_2):    

```

Comment:
This makes multiple shells, so maybe change the name.

Could also generalize this to take an inner radius plus layers:
```suggestion
def make_spherical_shells(inner_radius_W, layers):    
    '''
    Builds a set of nested spherical shells, with an inner_radius and
    multiple layers each with a material definition and a thickness
    
    inputs
    =====
    inner_radius : the radius of the inner surface of the first shell
    layers : list of tuples where each tuple is (material, thickness)
    '''
```

Reply:
I think materials will need to be an input as this function is also assigning materials to the geometry. I'm wondering if this can be done with one set of radii/thickness inputs and the materials from TwoLayers_Materials.

```suggestion
def make_spherical_shell(radii, materials):
    '''
    Takes a list of materials and radii (outermost to innermost)
    and creates an OpenMC Geometry object.
    
    element_list = list of elements (str) in order from radially outermost to innermost
    radii = list of the radii (outermost to innermost) that serve as the boundaries of the 
    spherical shells; must be in the same order as the specified materials
    '''
    cells = []
    for index, material in enumerate(materials):
        outer_sphere = openmc.Sphere(r = radii[index]) # add boundary_type here
        inner_sphere = openmc.Sphere(r = radii[index + 1]) 
        element_region = +inner_sphere & -outer_sphere
        element_shell = openmc.Cell(fill = materials[index], region = element_region)
        cells.append(element_shell)
        if material == materials[-1]:
            void = openmc.Cell(fill = None, region = -inner_sphere) 
            cells.append(void)
    geometry = openmc.Geometry(cells)  
    return geometry    

```

Reply:
I agree that the materials are necessary.  My version had them bound in the `layers`.  This kind of data structure can lead to clarity because the data gets bound together in instructive ways.  An example of calling it might be:
```
inner_radius = 10.0
# material_a and material_b are OpenMC Materials
layers = [(material_a, 5.0), (material_b, 4.0)]
geometry = make_spherical_shells(inner_radius, layers)
```

The function may be:
```
def make_spherical_shells(inner_radius, layers):

    inner_sphere = openmc.Sphere(inner_radius)
    cells = [openmc.Cell(region=-inner_sphere)]
    for (material, thickness) in layers:
        outer_radius = inner_radius + thickness
        outer_sphere = openmc.Sphere(outer_radius)
        cells.append(openmc.Cell(fill = material, region = inner_sphere & -outer_sphere)
        inner_radius = outer_radius
        inner_sphere = outer_sphere
    cells.append(openmc.Cell(region = outer_sphere)
    geometry = openmc.Geometry(cells)

    return geometry
```

(There may be minor syntax errors, but this is the basic idea)